Add feature request and bug report templates

ligurio opened the pull request:
Closes #5642

Totktonada commented:
See, we can direct a user with a question or a subject to discussion right after clicking on the 'New issue' button: https://github.com/curl/curl/issues/new/choose. That looks convenient.

ligurio commented:
Agree, it is very convenient. So I have updated PR and added config with redirection to a page with commercial support and page to create new discussion.

Totktonada commented on .github/ISSUE_TEMPLATE/config.yml:
```diff
@@ -0,0 +1,8 @@
+blank_issues_enabled: false
```
I would not block this option, because personally I sometimes file issues for refactoring, creating a tool or some raw ideas.

ligurio replied:
it seems you get it wrong

> The blank_issues_enabled option allows you to show or hide the Create a blank issue choice when users select New issue in your repository. 
https://github.blog/changelog/2019-10-28-new-issue-template-configuration-options/

I think it is better to hide "Create a blank issue".

Totktonada replied:
Seems no. I want to have ability to create an issue, which is not a bug report and not a feature request. And I would want it being an external man.

ligurio replied:
I believe it happens rarely and it is better to choose issue with template and erase text.
I'm afraid that with available option "Create a blank issue" users will prefer it than issues with templates, because filling all required fields in templates requires more strength.

Totktonada replied:
I guess an external user will be unable to add or removed labels.

ligurio replied:
Well, decided to keep option enabled.

Totktonada replied:
Thanks! Let's return here later, if we'll see any negative impact.

Totktonada commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,25 @@
+
+Add any other context or screenshots about the feature request here.
+
+**Definition of Done**
```
Maybe write someting in this section like we do for all other ones?

ligurio replied:
Yeah, but I don't know how to describe it. Will do.

ligurio replied:
added "A clear and concise understanding of expectations that the current feature request must meet in order to be released to users."

Totktonada commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,29 @@
+
+A clear and concise description of what the bug is.
+
+* Tarantool version (`tarantool --version`):
```
Maybe it worth to add `| head -n 1`.

Totktonada replied:
Or leave just 'Tarantool version:'.

igormunkin replied:
Sometimes it's necessary to see the build flags, so IMHO stripping this info is inconvenient for postmortem analysis.

ligurio replied:
`tarantool --version` contains flags, major, minor versios and part of commit hash, target and compiler. I think that all that information is valuable for us.

Totktonada replied:
So it should be like the following:

```
<details>
<summary>Tarantool version</summary>

<place `tarantool --version` output here>
</details>

OS: ...
```

The separation from the item-per-line `key: value` list is because the output is expected to be multiline. The `<details/>` tag is because the output is relatively big and mostly the similar.

ligurio replied:
discussed in private conversation, and now it looks like this https://gist.github.com/ligurio/8d8875b7c283b4f7ebdc103e1c777fe8

Totktonada replied:
> The separation from the item-per-line `key: value` list is because the output is expected to be multiline.

It was resolved.

> The `<details/>` tag is because the output is relatively big and mostly the similar.

It was declined by Sergey.

Okay, let's try and look.

Totktonada commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,29 @@
+
+**Steps to reproduce**
+
+1. Go to '...'
+1. Click on '....'
+1. Scroll down to '....'
```
I would give something more tarantool specific: those steps looks like we develop a GUI application. E.g.:

----

1. Run two tarantool instances iteractively.
2. Type '...' in console 1.
3. Type '...' in console 2.

Or:

Reproducer script:

```lua
#!/usr/bin/env tarantool

<your code here>
```

How to run:

Just type `tarantool test.lua`.

----

Real-world examples: #5559, #4294, #4114, #4106.

ligurio replied:
Done.

Totktonada commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,25 @@
+---
+name: Feature request
+about: Suggest a feature for the project
+labels: feature, incoming
```
It is unclear how 'incoming' and 'processed' issues are different. Confirmation that we're interested in the feature in theory? Set assignee and precise milestone? Since we don't have such a triage process now, I would not add any label of this kind.

We can return here after some discussions with @kyukhin and @Mons.

ligurio replied:
label `incoming` was requested by @Mons in a private conversation

Mons replied:
Agree. We building now such a process, so this label will be useful

Totktonada replied:
So we should set it for both feature requests and bug reports?

Totktonada replied:
Created the label so.

ligurio replied:
Mons requested label `incoming` for feature requests only.

Totktonada replied:
As I heard, the motivation was 'the team anyway catches bugs'. In fact, it is not so. Let's look at the [crashes](https://github.com/tarantool/tarantool/issues?q=is%3Aopen+is%3Aissue+label%3Acrash) for example: 43 open issues. And 221 [bugs](https://github.com/tarantool/tarantool/issues?q=is%3Aopen+is%3Aissue+label%3Abug+) at whole. Our priorities are quite dependent of customers and plans around valuable features. In fact, 'bug' label does not meant that it is always has higher priority than a feature. Sad, but now we're in such a reality.

Despite triage process we'll setup, all new issue should go though it.

So I'll cast @Mons and @kyukhin to decide here as owners of the future process.

If they will agree, let's resolve the discussion with the decision and go ahead. Or just resolve, go ahead and ask colleagues for feedback later. Up to you.

Totktonada commented:
That looks nice in general.

Totktonada reviewed (approved):
LGTM with one complain from my side regarding the future triage process. Anyway, it is up to @Mons and @kyukhin.
